File: app/src/main/java/com/termux/x11/controller/inputcontrols/InputEventHandler.java
package com.termux.x11.controller.inputcontrols;

import android.content.Context;
import com.termux.x11.ILorieView;

/**
 * Event handler interface for input events.
 */
public interface InputEventHandler {
    void onKeyEvent(int keycode, boolean isDown);
    void onPointerMove(int dx, int dy);
    void onPointerButton(int button, boolean isDown);
}

File: app/src/main/java/org/github/ewt45/winemulator/inputcontrols/InputEventHandler.kt
package org.github.ewt45.winemulator.inputcontrols

import android.view.KeyEvent

/**
 * Interface for handling input events from virtual controls
 */
interface InputEventHandler {
    /**
     * Handle a key event with full KeyEvent information
     * This supports ACTION_MULTIPLE events for continuous key presses (like WASD movement)
     * @param event The Android KeyEvent containing keycode, action, and repeat count
     */
    fun onKeyEvent(event: KeyEvent)

    /**
     * Handle pointer movement
     * @param dx Change in X coordinate
     * @param dy Change in Y coordinate
     */
    fun onPointerMove(dx: Int, dy: Int)

    /**
     * Handle pointer button event
     * @param button X11 button number (1=left, 2=middle, 3=right, 4=scroll up, 5=scroll down)
     * @param isDown True if pressed, false if released
     */
    fun onPointerButton(button: Int, isDown: Boolean)
}
